lib-rca/src/main/java/com/facebook/openwifi/librca/inputs/Configs.java

```diff
@@ -0,0 +1,174 @@
+	/** Maximum acceptable jitter (ms) */
+	public final int maxJitterThreshold;
+	/** Maximum acceptable disconnection rate (disconnetions per hour) */
+	public final int maxDisconnectionRate;
+
+	// High Level metrics parameters
+	/** Minimum acceptable tx rate (Mbps) */
+	public final double minTxRate;
+	/** Maximum acceptable Packet Error Rate (PER) (units are %) */
+	public final double maxPer;
+	/** Minimum acceptable idle airtime (units are %) */
+	public final double minIdleAirtime;
+	/** Maximum acceptable number of clients for one radio */
+	public final int maxNumClients;
+
+	// Low Level metrics parameters
+	/** Minimum acceptable RSSI (dBm) */
+	public final int minRssi;
+	/** Maximum acceptable noise (dBm) */
+	public final int maxNoise;
+	/** Maximum acceptable intf airtime (units are %) */
+	public final double maxIntfAirtime;
+	/** Maximum acceptable number of neighbors */
+	public final int maxNumNeighbors;
+	/** Minimum acceptable client bandwidth (MHz) for non-2G bands / */
+	public final int minClientBandwidth;
+	/** Minimum acceptable Access Point (AP) bandwidth (MHz) for non-2G bands */
+	public final int minApBandwidth;
+	/** Minimum acceptable self airtime ratio (units are %) */
+	public final double minSelfAirtimeRatio;
+	/** Maximum acceptable tx dropped ratio (units are %) */
+	public final double maxTxDroppedRatio;
+
+	/** Default constructor */
+	public Configs() {
+		this.detectionWindow = DEFAULT_DETECTION_WINDOW;
+
+		this.minEstimatedThroughput = DEFAULT_MIN_ESTIMATED_THROUGHPUT;
+		this.throughputAggregationPercentile = DEFAULT_THROUGHPUT_AGGREGATION_PERCENTILE;
+		this.maxLatencyThreshold = DFEAULT_MAX_LATENCY_THRESHOLD;
+		this.maxJitterThreshold = DEFAULT_MAX_JITTER_THRESHOLD;
+		this.maxDisconnectionRate = DEFAULT_MAX_DISCONNECTION_RATE;
+
+		this.minTxRate = DEFAULT_MIN_TX_RATE;
+		this.maxPer = DEFAULT_MAX_PER;
+		this.minIdleAirtime = DEFAULT_MIN_IDLE_AIRTIME;
+		this.maxNumClients = DEFAULT_MAX_NUM_CLIENTS;
+
+		this.minRssi = DEFAULT_MIN_RSSI;
+		this.maxNoise = DEFAULT_MAX_NOISE;
+		this.maxIntfAirtime = DEFAULT_MAX_INTF_AIRTIME;
+		this.maxNumNeighbors = DEFAULT_MAX_NUM_NEIGHBORS;
+		this.minClientBandwidth = DEFAULT_MIN_CLIENT_BANDWIDTH;
+		this.minApBandwidth = DEFAULT_MIN_AP_BANDWIDTH;
+		this.minSelfAirtimeRatio = DEFAULT_MIN_SELF_AIRTIME_RATIO;
+		this.maxTxDroppedRatio = DEFAULT_MAX_TX_DROPPED_RATIO;
+	}
+
+	private static void validatePositive(String varName, int value) {
+		if (value <= 0) {
+			throw new IllegalArgumentException(varName + " must be positive.");
+		}
+	}
+
+	private static void validatePositive(String varName, double value) {
+		if (value <= 0) {
+			throw new IllegalArgumentException(varName + " must be positive.");
+		}
+	}
+
+	private static void validatePercentile(String varName, double value) {
+		if (value < 0 || value > 100) {
+			throw new IllegalArgumentException(
+				varName + " must be between 0 and 100 inclusive.");
+		}
+	}
+
+	public void validate() {
```

Comment:
- add comment
- also, you might want to consider returning a list of errors instead, as throwing will only let you show the first error